assets/js/app.js

```diff
@@ -51,26 +53,42 @@ function App({ uniqueId, socket }) {
             socket.emit('ClientReady', {
                 token: token
             });
+            setupTokenRefresh();
         }
 
         // Listen for token events.
         socket.on('Token', (data) => {
             token = data.token;
             storeToken(token, data.expire);
+            setupTokenRefresh();
 
             // Signal that the client is ready.
             socket.emit('ClientReady', {
                 token: token
             });
         });
 
+        // Listen for refreshed token events.
+        socket.on('RefreshedToken', (data) => {
+            token = data.token;
+            storeToken(token, data.expire);
+            setupTokenRefresh();
+
```

Comment:
Should we add the possibility to refresh configuration and translation on token refresh? Not sure, just an idea.

Reply:
I think we should just send new configuration if it is changed. But that is not relevant for token refresh, I think.